sound/soc/bcm/audioinjector-pi-soundcard.c

```diff
@@ -158,9 +158,10 @@ static int audioinjector_pi_soundcard_probe(struct platform_device *pdev)
 			}
 	}
 
-	if ((ret = devm_snd_soc_register_card(&pdev->dev, card))) {
-		dev_err(&pdev->dev, "snd_soc_register_card failed (%d)\n", ret);
-	}
+	if (ret = devm_snd_soc_register_card(&pdev->dev, card))
+		if (ret != -EPROBE_DEFER)
```

Comment:
Thanks for the patch.

The preference seems to be for dev_err_probe now, as it also stores the probe reason.
https://elixir.bootlin.com/linux/latest/source/drivers/base/core.c#L4709